amazon-speech-api/amazonTranscribeHarris.js

```diff
@@ -35,40 +118,100 @@ function startTranscriptionJob(transcriptionObject) {
     OutputBucketName: transcriptionObject.outputBucketName,
     TranscriptionJobName: transcriptionObject.jobName, /* required */
     Settings: {
-      // ChannelIdentification: false,
-      // MaxSpeakerLabels: 0,
-      // ShowSpeakerLabels: false
+      VocabularyName: vocabularyObject.VocabularyName // To set the vocabulary
     }
   }
-    transcribeservice.startTranscriptionJob(params, function(err, data) {
-    if (err) console.log(err, err.stack); // an error occurred
-    else     console.log(data);           // successful response
-      });
-    };
+  transcribeservice.startTranscriptionJob(params, callback);
+};
+
+function getTranscriptionJob(callback) {
+  transcribeservice.getTranscriptionJob({
+    TranscriptionJobName: transcriptionObject.jobName
+  }, callback);
+}
 
-function getTranscriptionJob(outputJson){
-  request(outputJson, { json: true }, (err, res, body) => {
-    if (err) { return console.log(err); }
-      console.log(body.results.transcripts);
-  });
+function getFromS3Bucket(bucket, key, callback) { 
+  s3.getObject({
+    Bucket: bucket,
+    Key: key
+  }, callback);
 }
 
-function transcribe(transcriptionObject, newTranscribe=true) {
-  this.transcriptionObject = transcriptionObject;
-  if (newTranscribe == false){
-    startTranscriptionJob(transcriptionObject);
-  } 
+function transcribe(transcriptionObject, vocabularyObject, newTranscribe=true, callback) {
+	function sendTranscription(){
+		if (fetchedVocabulary){
+			getVocabulary(vocabularyObject.VocabularyName,(err,data)=> {
+				if (err) {console.log(err, err.stack)} // an error occurred
+				else {
+				  	const state = data.VocabularyState;
+				  	console.log('The current state of the vocabulary is',state);
+				  	if (state === 'READY' || state === 'FAILED'){
+				  		if (state === 'FAILED') {
+				  			return console.log(data);
+				  		}
+				  		console.log('===Starting Transcription Job');
+				  		return startTranscriptionJob(transcriptionObject, callback);
+				  	} else {setTimeout(sendTranscription,10000)};
+				}
+			})
+		} else{
+			createBucket(transcriptionObject.createBucket, transcriptionObject.file);
+		    createVocabulary(vocabularyObject);
+		    fetchedVocabulary = true;
+		    setTimeout(sendTranscription,10000)
+		}
+	}
+	if (newTranscribe == true){
```

Comment:
i generally prefer `===` over `==`.  Triple equal matches on type AND value but double equal matches on only value.  https://codeburst.io/javascript-double-equals-vs-triple-equals-61d4ce5a121a